core/mempool/src/lib.rs

```diff
@@ -83,14 +87,22 @@ impl Pool {
     }
 
     pub fn add_transaction(&self, transaction: SignedTransaction) -> Result<(), String> {
-        if let Ok(Some(_)) = self
-            .storage
-            .write()
-            .expect(POISONED_LOCK_ERR)
-            .transaction_address(&transaction.get_hash())
         {
-            return Ok(());
+            // validate transaction
+            let mut guard = self.storage.write().expect(POISONED_LOCK_ERR);
+            let last_block_index = guard.blockchain_storage_mut().best_block().unwrap().unwrap().index();
+            if last_block_index > 0 {
+                let old_tx_nonce = *guard.tx_nonce(last_block_index).unwrap().unwrap();
+                if transaction.body.get_nonce() <= old_tx_nonce {
+                    return Ok(());
```

Comment:
I am not sure I understand how this logic works.
Transaction nonce is increased by the transaction sender every time they send a transaction. So there is no such thing as maximum nonce of the block (`old_tx_nonce`), there is maximum nonce per sender per block.
If we compare maximum transaction nonce of block X with transaction nonce submitted to block X+1 then we will be discarding otherwise valid transactions. 

What am I missing?

Reply:
That was a mistake! I somehow misunderstood how nonce works. It's fixed now